### ISSUE TITLE
0.8.0 id=-1 is returned when total count < topk

### ISSUE THREAD

**Issue by sunyaofei** [milvus-io/milvus#2203]
**Describe the bug**
for example: 
1. insert 100 record to milvus
2. search with topk=200, i got 100 ids followed by 100 `-1`s, as below
```json
[
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118200,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118000,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118200,
  1588236398893118200,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  1588236398893118000,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1,
  -1
]
```

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**
just as 0.6.0, return only 100 ids

**Environment details**
milvus 0.8.0, java sdk

**Screenshots**

**Additional context**

**Comment by sunyaofei** [comment]
returned distance has the same problem

**Comment by tinkerlin** [comment]
When the number of results is less than topk, the insufficient part will be filled with -1, which is an expected behavior.
@sunyaofei

**Comment by sunyaofei** [comment]
> When the number of results is less than topk, the insufficient part will be filled with -1, which is an expected behavior.
> @sunyaofei

what is the consideration of this strategy ? why not retain the strategy as 0.6.0 ?
now we have to filter useless result with extra code

**Comment by tinkerlin** [comment]
I have reproduced this problem. There are indeed different behaviors between 060 and 080. We are working hard to find code changes and redefine this behavior.

**Comment by JinHai-CN** [comment]
@scsven As we discussed, this issue should be solved in SDK.

**Comment by sahuang** [comment]
This has been implemented in Java SDK and will be available with Milvus 0.9.0 Release.

**Comment by fishpenguin** [comment]
This has been implemented in Go SDK and will be available with Milvus 0.9.0 Release.

**Comment by BossZou** [comment]
This issue has been implemented in python SDK.
